server/oauth.ts

```diff
@@ -346,12 +347,27 @@ export const setLocalStateFromIdTokenOrUserCookie = (
 	// but not the other fields. This will allow the frontend to show the
 	// signed in menu, but not show the user's name or email.
 	const hasIdTokenOrUserCookie = idToken || req.cookies['GU_U'];
-	res.locals.identity = {
+
+	let identityIdClaim: string | undefined;
+	let nameClaim: string | undefined;
+	let emailClaim: string | undefined;
+
+	if (typeof idToken?.claims.legacy_identity_id === 'string') {
+		identityIdClaim = idToken?.claims.legacy_identity_id;
+	}
+	if (typeof idToken?.claims.name === 'string') {
+		nameClaim = idToken?.claims.name;
+	}
+	if (typeof idToken?.claims.email === 'string') {
+		emailClaim = idToken?.claims.email;
+	}
+
+	setIdentityLocalState(res, {
 		signInStatus: hasIdTokenOrUserCookie ? 'signedInRecently' : undefined,
-		userId: idToken?.claims.legacy_identity_id,
-		name: idToken?.claims.name,
-		email: idToken?.claims.email,
-	};
+		userId: identityIdClaim,
+		name: nameClaim,
+		email: emailClaim,
+	});
```

Comment:
this doesn't work because `idToken.claims` doesn't have the actual claims defined in the type. 
I am trying to assigning to a string so I guess typescript has to know it's going to be a string ?
```
   server/oauth.ts:354:10 - error TS2339: Property 'legacy_identity_id' does not exist on type 'JwtClaims | undefined'.

    354  const { legacy_identity_id: userId, name, email } = idToken?.claims;
                 ~~~~~~~~~~~~~~~~~~
    server/oauth.ts:354:38 - error TS2339: Property 'name' does not exist on type 'JwtClaims | undefined'.

    354  const { legacy_identity_id: userId, name, email } = idToken?.claims;
                                             ~~~~
    server/oauth.ts:354:44 - error TS2339: Property 'email' does not exist on type 'JwtClaims | undefined'.

    354  const { legacy_identity_id: userId, name, email } = idToken?.claims;
                                                   ~~~~~

```

Reply:
It looks like the TS error was just because `claims` could have been undefined; this should work:

```
	if (!idToken?.claims) {
		throw new Error('ID token is missing claims');
	}

	const { legacy_identity_id: userId, name, email } = idToken.claims;

	setIdentityLocalState(res, {
		signInStatus: hasIdTokenOrUserCookie ? 'signedInRecently' : undefined,
		userId,
		name,
		email,
	});
```

You may need to handle the `throw` upstream.